flocker/control/_protocol.py

```diff
@@ -548,11 +604,16 @@ def __init__(self, reactor, cluster_state, configuration_service, endpoint,
         :param endpoint: Endpoint to listen on.
         :param context_factory: TLS context factory.
         """
-        self.connections = set()
+        self._connections = set()
         self._reactor = reactor
         self._connections_pending_update = set()
         self._current_pending_update_delayed_call = None
         self._current_command = {}
+        self._last_recieved_generation = defaultdict(
+            lambda: _ConfigAndStateGeneration()
+        )
```

Comment:
scratch that, wrong thread.


On 6 June 2016 at 07:42, Li Ding <li.ding@clusterhq.com> wrote:

> i saw it once out of 4 times i ran it over the weekend
>
> On 6 June 2016 at 03:10, Richard Wall <notifications@github.com> wrote:
>
>> In flocker/control/_protocol.py
>> <https://github.com/ClusterHQ/flocker/pull/2811#discussion_r65866502>:
>>
>> >          self._reactor = reactor
>> >          self._connections_pending_update = set()
>> >          self._current_pending_update_delayed_call = None
>> >          self._current_command = {}
>> > +        self._last_recieved_generation = defaultdict(
>> > +            lambda: _ConfigAndStateGeneration()
>> > +        )
>>
>> typo "recieved"
>>
>> —
>> You are receiving this because you are subscribed to this thread.
>> Reply to this email directly, view it on GitHub
>> <https://github.com/ClusterHQ/flocker/pull/2811/files/329e316652b7dffa4daa491ee47069ff3f69eefa#r65866502>,
>> or mute the thread
>> <https://github.com/notifications/unsubscribe/ARmHhm7xcT_Ri32jtuS0O897V0Asth2qks5qI_H4gaJpZM4Ity89>
>> .
>>
>
>